fetch_toolchain

```diff
@@ -7,16 +7,17 @@
 
 set -euo pipefail
 
-readonly PREFIX_JS=/tmp/toolchains/llvm-js
-readonly PREFIX_EMS=/tmp/toolchains/emscripten
-readonly tmpDir=/tmp/dscripten-tmp
-readonly scriptDir=$(realpath $(dirname $0))
+readonly PREFIX=${PREFIX:-/tmp/toolchains}
+readonly PREFIX_JS=$PREFIX/llvm-js
+readonly PREFIX_EMS=$PREFIX/emscripten
+readonly tmpDir=${tmpDir:-/tmp/dscripten-tmp}
+readonly scriptDir=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)
 
-readonly MAKE="make -j`nproc`"
+readonly MAKE=(make -j"$(nproc)")
 
 function main
 {
-  mkdir -p $tmpDir
+  mkdir -p "$tmpDir"
```

Comment:
Allowing spaces in tmpDir and prefix is an invitation for trouble ; do we really want to allow this?

Reply:
Allowing stars/question marks/etc in tmpDir and prefix is also an invitation for trouble.

Basically, we would be littering the code with double quotes, to support some use case that we want to disallow anyway. I see no concrete benefit here, appart from being able to tick the box "proper bash syntax".

What about checking the values of tmpDir and PREFIX at startup? This is probably something we ought to do anyway.